packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts

```diff
@@ -0,0 +1,450 @@
+                    String.fromCharCode(index + 65),
+                    dataStoreRuntime,
+                    DirectoryFactory.Attributes,
+                );
+                const containerRuntime = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime);
+                const services: IChannelServices = {
+                    deltaConnection: containerRuntime.createDeltaConnection(),
+                    objectStorage: new MockStorage(),
+                };
+
+                sharedDirectory.initializeLocal();
+                sharedDirectory.connect(services);
+                return { containerRuntime, sharedDirectory };
+            });
+
+            const initialState: FuzzTestState = {
+                clients,
+                containerRuntimeFactory,
+                random: makeRandom(seed),
+            };
+
+            runSharedDirectoryFuzz(
+                generator,
+                initialState,
+                { saveOnFailure: true, filepath: getPath(seed) },
+                loggingInfo,
+            );
+        });
+    }
+
+    for (let i = 0; i < testCount; i++) {
+        const generator = take(100, makeOperationGenerator({ validateInterval: 10 }));
+        runTests(i, generator);
+    }
+});
+
+/**
+ * Validates that all shared directories in the provided array are consistent in the underlying keys/values
+ * and sub directories recursively.
+ * */
+function assertEventuallyConsistentDirectoryState(clients: Client[]): void {
+    const connectedClients = clients.filter((client) => client.containerRuntime.connected);
+    if (connectedClients.length < 2) {
+        // No two strings are expected to be consistent.
+        return;
+    }
+    const first = connectedClients[0].sharedDirectory;
+    for (const { sharedDirectory: second } of connectedClients.slice(1)) {
+        assertEventualConsistencyCore(first, second);
+    }
+}
+
+function assertEventualConsistencyCore(first: SharedDirectory, second: SharedDirectory) {
+    // Check number of keys.
+    assert.strictEqual(first.size, second.size, `Number of keys not same: Number of keys in ` +
+        `${first.id}: ${first.size} and in ${second.id}: ${second.size}`);
+
+    // Check key/value pairs in both directories.
+    for (const key of first.keys()) {
+        assert.strictEqual(first.get(key), second.get(key), `Key not found or value not matching ` +
+            `key: ${key}, value in dir ${first.id}: ${first.get(key)} and in ${second.id}: ${second.get(key)}`);
+    }
+
+    // Check for number of subdirectores with both directories.
+    assert.strictEqual(first.countSubDirectory(), second.countSubDirectory(),
+        `Number of subDirectories not same: Number of subdirectory in ` +
+        `${first.id}: ${first.countSubDirectory()} and in ${second.id}: ${second.countSubDirectory()}`);
+
+    // Check for consistency of subdirectores with both directories.
+    for (const [name, subDirectory1] of first.subdirectories()) {
```

Comment:
i'm not particularly familiar with the subdirectory API, but it seems like you either want to do a length assertion on size of subdirectories matching OR validate that there aren't subdirectories in the second that are in the first here.

Reply:
So, a subdirectory could have key, value pairs and sub directories. So, I am matching the size which is no. of keys in it and then check that all key value pairs match. And then I am checking number of sub directories and then recursively checking eventual consistency on those sub directories.